lib/openhab/core/emulate_hash.rb

```diff
@@ -0,0 +1,233 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+module OpenHAB
+  module Core
+    #
+    # Make an openHAB class acts like Ruby Hash.
+    #
+    module EmulateHash
+      include Enumerable
+      extend Forwardable
+
+      # The implementing class must implement #to_hash
+      def_delegators :to_hash,
+                     :any?,
+                     :compact,
+                     :compare_by_identity?,
+                     :deconstruct_keys,
+                     :each,
+                     :each_key,
+                     :each_pair,
+                     :each_value,
+                     :equals,
+                     :empty?,
+                     :filter,
+                     :flatten,
+                     :has_value?,
+                     :invert,
+                     :key,
+                     :keys,
+                     :length,
+                     :merge,
+                     :rassoc,
+                     :reject,
+                     :select,
+                     :shift,
+                     :size,
+                     :to_a,
+                     :to_h,
+                     :transform_keys,
+                     :transform_values,
+                     :values,
+                     :value?
+
+      # @!visibility private
+      def replace
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def to_hash
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def fetch(key)
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def store(key, value)
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def [](key)
+        fetch(key, nil)
+      end
+
+      # @!visibility private
+      def []=(key, value)
+        store(key, value)
+      end
+
+      # @!visibility private
+      def assoc(key)
+        to_hash.assoc(key.to_s)
+      end
+
+      # @!visibility private
+      def clear
+        replace({})
+      end
+
+      # @!visibility private
+      def merge!(*others, &block)
+        return self if others.empty?
+
+        new_config = to_h
+        others.each do |h|
+          new_config.merge!(h.transform_keys(&:to_s), &block)
+        end
+        replace(new_config)
+      end
+      alias_method :update, :merge!
+
+      # @!visibility private
+      def dig(*keys)
```

Comment:
you lost the `key, *keys`, which is important, because only the first key should be converted to a string